src/result_processor.c

```diff
@@ -1740,15 +1745,14 @@ dictType dictTypeHybridSearchResult = {
         numConsumed++;
         continue;
       } else if (rc == RS_RESULT_TIMEDOUT){
-          rm_free(consumed);
           if (rp->parent->timeoutPolicy == TimeoutPolicy_Return) {
-            // Switch to yield phase and return immediately on timeout.
-            // Note: This may leave some ready results in other upstreams unconsumed.
+            // continue processing other upstreams since they might have available results
             self->timedOut = true;
-            self->iterator = dictGetIterator(self->hybridResults);
-            rp->Next = RPHybridMerger_Yield;
-            return rp->Next(rp, r);
+            consumed[i] = true;
+            numConsumed++;
+            continue;
           } else { // Strict
+            rm_free(consumed);
             return rc;
```

Comment:
This early return is still problematic, from the same reason the other path was fixed